Sources/Sentry/SentryANRTrackingIntegration.m

```diff
@@ -70,23 +72,33 @@ - (BOOL)shouldBeDisabled:(SentryOptions *)options
 
 - (void)uninstall
 {
-    [self.tracker stop];
+    [self.tracker removeListener:self];
 }
 
 - (void)anrDetected
 {
-    [self.appStateManager
-        updateAppState:^(SentryAppState *appState) { appState.isANROngoing = YES; }];
+    NSString *message = [NSString stringWithFormat:@"App hanging for at least %li ms.",
+                                  (long)(self.options.appHangTimeoutInterval * 1000)];
+
+    NSArray<SentryThread *> *threads =
+        [SentrySDK.currentHub.getClient.threadInspector getCurrentThreadsWithStackTrace:YES];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"App Hanging"];
+    sentryException.mechanism = [[SentryMechanism alloc] initWithType:@"AppHang"];
+    sentryException.stacktrace = threads[0].stacktrace;
```

Comment:
`h`: Are we sure that the thread with index 0 is always the main thread? I'm not 100% sure about that, can you please double-check?

Reply:
Apples [task_threads](https://developer.apple.com/documentation/kernel/1537751-task_threads) documentation has no information on it. According to manual tests, the assumption that the first thread is the main thread looks right.

Reply:
An alternative would to look into each thread for the one that contains the `main` function as first frame

Reply:
We could store the `mainThreadID` in the `SentryThreadInspector`. We would need to run the following code on the mainThread. Important to notice is that if you run this on the main dispatch queue the mainThreadID is different. I'm not sure why. Then the `SentryThreadInspector` could ensure that the first thread is always the main thread or we do something different to ensure that we use the main thread stacktrace here.
```
SentryCrashThread mainThreadID = pthread_mach_thread_np(pthread_self());
```